City-of-Helsinki/drupal-hdbt-admin
# UHF-X Fix preview image 404

## What was done
* Added design selection manager to ease up the preview image handling.
* Converted design field widget and paragraph selection to set preview images as list in to drupalSettings.
* Fixed image previewer javascripts to use the drupalSettings lists instead of producing 404 on each missing preview image
* Fixed min-width for the preview image thumbnails.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the HDBT Admin theme
    * `composer require drupal/hdbt_admin:dev-UHF-X_fix_preview_image_404`
* Run `make drush-cr`

## How to test
* Go to add/edit any page with paragraphs
* Browse through the paragraphs and check that there's no 404 messages in console. Like the image below
![image](https://user-images.githubusercontent.com/1712902/199055660-6cab5edf-a7f8-4adc-abf0-544acad58650.png)
* Check also that the banner paragraph design selection works and doesn't print out 404 messages
![image](https://user-images.githubusercontent.com/1712902/199056395-e891b962-e374-472c-9814-0833ec2ee4c7.png)
